group/views.py

```diff
@@ -205,6 +225,77 @@ def edit(request, group_id):
                 if form.is_valid():
                     modified_group = form.save()
                     logger.info('Group: Modified group %s!' % modified_group.id)
-                    return HttpResponseRedirect('/group/detail/%s' % modified_group.id)
+                    return HttpResponseRedirect(reverse('group:detail', kwargs={'group_id': modified_group_id}))
         else:
             raise PermissionDenied
+
+##Announce
+@login_required
+def add_announce(request, group_id):
+    group = get_group(group_id)
+
+    if has_group_ownership(request.user, group) or has_group_coownership(request.user, group):
+        if request.method == 'POST':
+            form = AnnounceForm(request.POST)
+            if form.is_valid():
+                new_announce = form.save()
+                group.announce.add(new_announce)
+                logger.info('Announce: User %s add Announce %s!' % (request.user.username, new_announce.id))
+                return HttpResponseRedirect(reverse('group:detail', kwargs={'group_id': group_id}))
+            else:
+                return HttpResponseRedirect(reverse('group:detail', kwargs={'group_id': group_id}))
+    else:
+        raise PermissionDenied
+
+@login_required
+def delete_announce(request, announce_id, group_id):
+    group = get_group(group_id)
+
+    if has_group_ownership(request.user, group) or has_group_coownership(request.user, group):  
+        try:
+            get_announce(announce_id).delete()
+            return HttpResponseRedirect(reverse('group:detail', kwargs={'group_id': group_id}))
+        except get_announce(announce_id).DoesNotExist:
```

Comment:
`` except get_announce(announce_id).DoesNotExist:`` part is not necessary because ``get_announce`` has done that for you.